### ISSUE TITLE
curl: (3) URL rejected: No host part in the URL

### ISSUE THREAD

**Issue by LukeShortCloud** [dragoonDorise/EmuDeck#849]
I have read in many places that if EmuDeck fails to download then follow the guide below to (1) disable IPv6 and (2) set the Cloudflare DNS resolver. I did both and still run into `curl` issues. I have tried on both Wi-Fi and Ethernet. I have re-installed EmuDeck many times. I have re-installed my OS many times (winesapOS, based on Arch Linux). It does not appear to be an Internet issue.

https://github.com/dragoonDorise/EmuDeck/wiki/frequently-asked-questions#why-wont-emudeck-download

The first thing I noticed missing was the Steam ROM Manager. Using the "Easy" installer method, all of the Flatpaks get installed but anything that is an AppImage or tarball fails with the same error message from `curl`. Here's the full log entry:

```
$ less ~/emudeck/emudeck.log
install srm
Installing Steam Rom Manager
safeDownload()
- Steam Rom Manager
-
- /home/winesap/Emulation/tools/srm/Steam-ROM-Manager.AppImage
-
-
curl: (3) URL rejected: No host part in the URL
HTTP response code: 000
CURL exit code: 3
Steam Rom Manager download failed
```

I can confirm that the `/home/winesap/Emulation/tools/srm/` folder is empty.

In total, 5 items failed to download:

```
$ grep curl ~/emudeck/emudeck.log
curl: (3) URL rejected: No host part in the URL
curl: (3) URL rejected: No host part in the URL
curl: (3) URL rejected: No host part in the URL
curl: (3) URL rejected: No host part in the URL
curl: (3) URL rejected: No host part in the URL
```

Thanks for any help you can provide!

**Comment by Godsbane** [comment]
My bet, missing jq

**Comment by LukeShortCloud** [comment]
Thanks for your input, @Godsbane ! That appears to be the issue. I see this in the AppImage log file:

```
$ grep jq ~/emudeck/Emudeck.AppImage.log
/home/winesap/.config/EmuDeck/backend/functions/helperFunctions.sh: line 674: jq: command not found
/home/winesap/.config/EmuDeck/backend/functions/helperFunctions.sh: line 674: jq: command not found
```

This dependency seems to be handled via this PR which is merged into the `early` and `dev` branches (but not in `main` yet). I tried to install those other branches but it says I need to be a Patreon supporter or have an access token.

```
$ curl -L https://raw.githubusercontent.com/dragoonDorise/EmuDeck/early/install.sh | bash
```

https://github.com/dragoonDorise/EmuDeck/pull/830

I manually checked the Arch Linux dependencies listed in the PR: `steam jq zenity flatpak unzip bash fuse2 git rsync`. The only one I was missing was `jq`. After installing it and re-installing EmuDeck, everything works! Steam ROM Manager now loads and works as expected.

**Comment by LukeShortCloud** [comment]
For reference, this is what a correct `curl` download log entry looks like:

```
$ less ~/emudeck/emudeck.log
install srm
Installing Steam Rom Manager
safeDownload()
- Steam Rom Manager
- https://github.com/SteamGridDB/steam-rom-manager/releases/download/v2.4.17/Steam-ROM-Manager-2.4.17.AppImage
- /home/winesap/Emulation/tools/srm/Steam-ROM-Manager.AppImage
-
-
<OMITTED_WGET_DOWNLOAD_PROGRESS_OUTPUT>
HTTP response code: 200
CURL exit code: 0
Steam Rom Manager download successfully
renamed '/home/winesap/Emulation/tools/srm/Steam-ROM-Manager.AppImage.temp' -> '/home/winesap/Emulation/tools/srm/Steam-ROM-Manager.AppImage'
'/home/winesap/.config/EmuDeck/backend-icons/srm.png' -> '/home/winesap/.local/share/icons/emudeck/srm.png'
```

**Comment by LukeShortCloud** [comment]
There's logic in `main` today to handle Pacman dependency installations but it tries to install `libfuse2` which has been renamed to `fuse2` so the installation for all of the packages fails (which includes `jq`). The merged PR in the `early` branch already addresses that so we should be good to close this once it lands in the `main` branch.

**Comment by Godsbane** [comment]
Probably, we should break up each dependency so one doesn't stop the others
